packages/fhir/src/Adaptor.js

```diff
@@ -38,202 +57,219 @@ export function execute(...operations) {
+      state,
+      path,
+      params,
+      options
+    );
 
-    return request(url, options, 'POST').then(({ data }) =>
-      callback({ ...state, data })
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      resolvedPath,
+      { ...resolvedOptions, query: resolvedParams }
     );
+    return util.prepareNextState(state, response, callback);
   };
 }
 
 /**
- * Creates a transactionBundle for HAPI FHIR
+ * Creates a resource by sending a request to the baseURL defined in config
+ * The resource object doesn't need resourceType.
  * @public
- * @example
- * createTransactionBundle({
- *   resourceType: "Bundle",
- *   type: "transaction",
- *   entry: [
+ * @function
+ * @example <caption>Create a new patient</caption>
+ * create('Patient', {
+ *   name: [
  *     {
- *       fullUrl: "https://hapi.fhir.org/baseR4/Patient/592442",
- *       resource: {
- *         resourceType: "Patient",
- *         id: "592442",
- *         name: [{ given: "Caleb", family: "Cushing" }],
- *       },
- *       request: {
- *         method: "POST",
- *         url: "Patient",
- *       },
+ *       use: 'official',
+ *       family: 'La Paradisio',
+ *       given: ['Josephine', 'Nessa'],
  *     },
  *   ],
  * });
- * @function
- * @param {object} params - data to create the new transaction
+ * @param {string} resourceType - The resource type to create
+ * @param {object} resource - The resource to create
+ * @param {object} params - (Optional) FHIR parameters to control and configure resource creation
  * @param {function} callback - (Optional) callback function
  * @returns {Operation}
+ * @state {FHIRHttpState}
  */
-export function createTransactionBundle(params, callback) {
-  return state => {
-    const [resolvedParams] = expandReferences(state, params);
-
-    const { baseUrl, apiPath, authType, token } = state.configuration;
-
-    const url = `${baseUrl}/${apiPath}`;
-    const auth = `${authType} ${token}`;
+export function create(resourceType, resource, params, callback = s => s) {
+  return async state => {
+    const [resolvedResourceType, resolvedResource, resolvedParams] =
+      expandReferences(state, resourceType, resource, params);
 
-    const options = {
-      body: {
-        resourceType: 'Bundle',
-        type: 'transaction',
-        entry: resolvedParams,
-      },
-      auth,
+    const opts = {
+      ...resolvedParams,
+      body: { ...resolvedResource, resourceType: resolvedResourceType },
     };
 
-    return request(url, options, 'POST').then(({ response, data }) =>
-      handleResponse(response, data, state, callback)
+    const response = await util.request(
+      state.configuration,
+      'POST',
+      resolvedResourceType,
+      opts
     );
+    return util.prepareNextState(state, response, callback);
   };
 }
 
 /**
- * Get a resource in a FHIR system
+ * Creates a transactionBundle for HAPI FHIR
```

Comment:
Small change, can we remove "HAPI FHIR". And can we add a bit more context, like `Create a transactionBundle to process multiple requests at once` or something. I want users who don't know FHIR to be able to understand what this is

Reply:
Done ✅